contrib/autodump.jl

```diff
@@ -0,0 +1,163 @@
+using Base.Filesystem
+using Dates
+
+const TIMEOUT = 2*60*60 # seconds
```

Comment:
Instead of having a `TIMEOUT` hardcoded like this, can we have this react to the buildbot's exit signal?  We can specify it to set `SIGTERM`, then respond to that here?

Reply:
I thought about it, but I actually think I would prefer not to use the buildbot's timeout mechanism. If this dumping doesn't complete for whatever reason, I think we should keep it untouched to allow for manual investigation of whatever weird state the kernel is in.